pkg/internal/day/duration.go

```diff
@@ -20,7 +20,13 @@ const (
 
 // Duration acts like a time.Duration with support for "d" unit
 // which is used for specifying number of days in duration.
-type Duration time.Duration
+// It stores the text of duration while parsing, which can be retrieved via Text().
+// This can be useful when we need to know the num of days user wanted, since
```

Comment:
Is internal `text` being used anywhere? I did a quick scan and didn't find any usage of it.
If not being used I'd recommend avoid creating new Duration type.

Reply:
Yes, we use it in `Config.Name()`. It is important, otherwise the downsampling schemas will be created as `ds_24h` if user provided `1d` as input.

Reply:
Yeah I was thinking the same. It seems the only way out is to always use `minute` granularity?

Reply:
Yeah. It seems we are better with just using Duration String(). For example: `time.Second * 3600` and `time.Minute * 60` will both print `1h0m0s` which is what we want.

Reply:
> I asked it [here](https://github.com/timescale/promscale/pull/1808/files#r1060615105) but is there a problem if a user creates 2 configs, one with 24h and another with 1d?

Yes, that is a valid point. I am not sure how to work around this. Doing `duration.String()` is not good, since then we will have schemas like `ds_1h0m0s` which is not tidy. We cannot choose to convert everything into hours or minutes, otherwise it hurts readability. So I left this as the user gave input. Any other suggestions please?

Reply:
>Doing duration.String() is not good, since then we will have schemas like ds_1h0m0s which is not tidy

Is that problematic in any way? Mostly asking b/c I want us to avoid Duration wrapper

Reply:
can the user do `1.5d` and `36h`?

Reply:
I believe this will produce a desired standardized output regardless of how the user specifies the input.

here is the output for the samples given:

```
1d
1d12h
12h
12h15m
15m30s
1d12h15m30s
```

```go
package main

import (
	"fmt"
	"time"
)

func main() {
        // sample tests
	fmt.Println(String(time.Hour * 24))
	fmt.Println(String(time.Hour * 36))
	fmt.Println(String(time.Hour * 12))
	fmt.Println(String(time.Hour*12 + time.Minute*15))
	fmt.Println(String(time.Minute*15 + time.Second*30))
	fmt.Println(String(time.Hour*36 + time.Minute*15 + time.Second*30))
}

func String(d time.Duration) string {
	const day = int64(time.Hour * 24)

	remainder := int64(d)
	days := remainder / day
	remainder = remainder % day
	hours := remainder / int64(time.Hour)
	remainder = remainder % int64(time.Hour)
	mins := remainder / int64(time.Minute)
	remainder = remainder % int64(time.Minute)
	secs := remainder / int64(time.Second)

	display := ""
	if days != 0 {
		display = fmt.Sprintf("%dd", days)
	}
	if hours != 0 {
		display = fmt.Sprintf("%s%dh", display, hours)
	}
	if mins != 0 {
		display = fmt.Sprintf("%s%dm", display, mins)
	}
	if secs != 0 {
		display = fmt.Sprintf("%s%ds", display, secs)
	}
	return display
}
```

Reply:
We need to have a deterministic algorithm for how we name schemas. There may be situations (hopefully not) where we have to reverse engineer schema names to figure out what they were.

Reply:
So we could use the standard library's `time.Duration` for parsing, and use the function above (or similar) for generating the schema name.

Reply:
my vote: if we want to support `d` units in input, use `duration.Day` BUT use a standardize output format like the above instead of the raw user input.

Reply:
If we have a deterministic algorithm for the schema names, then we don't have to worry about detecting the case where two different schema names have the same interval. That's a big win IMO